src/Symfony/Component/Mime/Part/TextPart.php

```diff
@@ -63,8 +65,8 @@ public function __construct($body, ?string $charset = 'utf-8', string $subtype =
         if (null === $encoding) {
             $this->encoding = $this->chooseEncoding();
         } else {
-            if ('quoted-printable' !== $encoding && 'base64' !== $encoding && '8bit' !== $encoding) {
-                throw new InvalidArgumentException(sprintf('The encoding must be one of "quoted-printable", "base64", or "8bit" ("%s" given).', $encoding));
+            if (!\in_array($encoding, self::DEFAULT_ENCODERS) && !\array_key_exists($encoding, self::$encoders)) {
+                throw new InvalidArgumentException('The encoding must be one of "quoted-printable", "base64", or "8bit". In case you are trying to use a custom encoder, please add it first.');
```

Comment:
Shall we just list _all_ available encodings? I don't think the second sentence is necessary then.

Reply:
Sure, sounds good. 
I'll change it.